packages/path_provider/path_provider_windows/lib/src/guid.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:typed_data';
+
+/// Representation of the Win32 GUID struct.
+// For the layout of this struct, see
+// https://learn.microsoft.com/en-us/windows/win32/api/guiddef/ns-guiddef-guid
+@Packed(4)
+base class GUID extends Struct {
+  /// Navite Data1 field.
+  @Uint32()
+  external int data1;
+
+  /// Navite Data2 field.
+  @Uint16()
+  external int data2;
+
+  /// Navite Data3 field.
+  @Uint16()
+  external int data3;
+
+  /// Navite Data4 field.
+  // This should be an eight-element byte array, but there's no such annotation.
+  @Uint64()
+  external int data4;
+
+  /// Parses a GUID string, with optional enclosing "{}"s and optional "-"s,
+  /// into data.
+  void parse(String guid) {
+    final String hexOnly = guid.replaceAll(RegExp(r'[{}-]'), '');
```

Comment:
`package:win32`'s `GUID.parse` requires the `{}`s and `-`s. Should we do that as well?

Reply:
I'm removing those symbols anyway to make the implementation simple, and since https://en.wikipedia.org/wiki/Universally_unique_identifier#Textual_representation says they are all valid I figured I would just accept all of them and relaxed the API accordingly. I didn't look at the `win32` implementation so I don't know if there's a reason they are stricter; the only one I could think of would be efficiency, and I'm pretty confident that it's never going to matter in this use case. (If efficiency were ever actually important the way to go would be to make an API that's not string-based in the first place, and constants that built the GUIDs directly.)

Reply:
Sounds good to me, thanks for the thoughtful reply